### PR TITLE
feat!: implement Minuit2 optimization functionality 

### PR DESCRIPTION
**Pull request by spflueger**

### CODE REVIEW

File: tensorwaves/estimator.py [hunk 1]
```diff
@@ -1,11 +1,49 @@
...
+        self.__model = model
+        self.__dataset = dataset
+
+    def __call__(self) -> float:
+        props = self.__model(self.__dataset)
+        logs = tf.math.log(props)
+        log_lh = tf.reduce_sum(logs)
+        return -log_lh.numpy()
+
+    def gradient(self) -> np.ndarray:
```
Comment (redeboer) [inline tensorwaves/estimator.py:31]: There's no issue on the missing gradient yet, right?

File: tensorwaves/estimator.py [hunk 2]
```diff
@@ -1,11 +1,49 @@
...
-        super(UnbinnedLL, self).__init__(name="UnbinnedLL")
+    @property
+    def parameters(self) -> dict:
+        return self.__model.parameters
 
-    def call(self, y_true, y_pred):
-        logs = tf.math.log(y_pred)
-        ll = tf.reduce_sum(logs)
-        return ll
+    def update_parameters(self, new_parameters: dict) -> None:
```
Comment (redeboer) [inline tensorwaves/estimator.py:48]: Why not a setter?

File: tensorwaves/optimizer/minuit.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
+"""Minuit2 adapter to the iminuit package."""
```
Comment (redeboer) [inline tensorwaves/optimizer/minuit.py:1]: Intersphinx reference to `iminuit`'s API?
https://iminuit.readthedocs.io/en/latest/

File: tensorwaves/estimator.py (hunk 2 above)
Comment (spflueger) [reply tensorwaves/estimator.py:48]: To me the `update_parameters` makes it clearer that the you are not replacing parameters but merely updating some of their values

File: tensorwaves/optimizer/minuit.py (hunk 3 above)
Comment (spflueger) [reply tensorwaves/optimizer/minuit.py:1]: True

File: tensorwaves/estimator.py (hunk 1 above)
Comment (spflueger) [reply tensorwaves/estimator.py:31]: Good point. I will open an issue and remove this code.

Comment (spflueger) [reply tensorwaves/estimator.py:31]: See #64
